src/store/contact/contactSlice.ts

```diff
@@ -0,0 +1,55 @@
+// This slice is responsible for managing the contacts in the store. It is used to add, update, and remove contacts.
+// Whenever there is a new conversation, the contacts are added to the store.
+// It also manages the availability status of the contacts
+import { createSlice, createEntityAdapter } from '@reduxjs/toolkit';
+import { Contact } from '@/types/Contact';
+import { Conversation } from '@/types/Conversation';
+
+export const contactAdapter = createEntityAdapter<Contact>({
+  selectId: contact => contact.id,
+});
+
+const initialState = contactAdapter.getInitialState({});
+
+const contactSlice = createSlice({
+  name: 'contact',
+  initialState,
+  reducers: {
+    clearAllContacts: contactAdapter.removeAll,
+    addContacts: (state, action) => {
+      const { conversations } = action.payload;
+      const contacts = conversations.map((conversation: Conversation) => conversation.meta.sender);
+      contacts.map((contact: Contact) => {
+        contactAdapter.upsertOne(state, contact);
+        return contact.id;
+      });
```

Comment:
I'm not sure why we need the conversation object on a contactSlice. It should be synthesized in the action before you pass it here.